pdal/PointView.hpp

```diff
@@ -304,6 +304,20 @@ class PDAL_DLL PointView : public PointContainer
     KD3Index& build3dIndex();
     KD2Index& build2dIndex();
 
+    template <typename Compare>
+    void stableSort(Compare compare)
+    {
+        auto order = m_index;
+        std::stable_sort(
+            order.begin(),
+            order.end(),
+            [this, &compare](const PointId a, const PointId b)
+            {
+                return compare(PointRef(*this, a), PointRef(*this, b));
```

Comment:
Not a big deal, but you could avoid the PointRef altogether and accept PointId's. There's already a comparator for around for that, but it also takes a dimension, which you could std::bind if you wanted.